usb/usb_host_msc/private_include/msc_common.h

```diff
@@ -45,14 +45,15 @@ typedef struct msc_host_device {
  * @brief Trigger a BULK transfer to device: zero copy
  *
  * Data buffer ownership is transferred to the MSC driver and the application cannot access it before the transfer finishes.
- * The buffer must be DMA capable, as it is going to be accessed by USB DMA.
+ * This function is true zero-copy only if the passed data buffer is in DMA capable memory, as the USB Host uses DMA transfers.
+ * If the buffer is NOT DMA capable, an intermediate DMA capable buffer is allocated and used for the transfer, resulting in memory coping.
  *
- * This function significantly improves performance with usage of Virtual File System, which creates a intermediate buffer for each opened file.
- * The intermediate VFS buffer is then used for USB transfers too, which eliminates need of 2 large buffers and unnecessary copying of the data.
+ * This function significantly improves performance with usage of Virtual File System, which creates its own buffer for each opened file.
```

Comment:
AFAIK, `Virtual File System` does not maintain the buffer itself, the buffer is maintained by newlib with the `fopen` function, right?

Reply:
Yes, thank you! I'll update the docs